src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/RazorLanguageServerClient.cs

```diff
@@ -0,0 +1,75 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using System.Diagnostics;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.LanguageServer.Client;
+using Microsoft.VisualStudio.Threading;
+using Microsoft.VisualStudio.Utilities;
+
+namespace Microsoft.VisualStudio.LanguageServerClient.Razor
+{
+    [Export(typeof(ILanguageClient))]
+    [ContentType(RazorLSPContentTypeDefinition.Name)]
+    internal class RazorLanguageServerClient : ILanguageClient
+    {
+        public string Name => "Razor Language Server Client";
+
+        public IEnumerable<string> ConfigurationSections => null;
+
+        public object InitializationOptions => null;
+
+        public IEnumerable<string> FilesToWatch => null;
+
+        public event AsyncEventHandler<EventArgs> StartAsync;
+        public event AsyncEventHandler<EventArgs> StopAsync
+        {
+            add { }
+            remove { }
+        }
+
+        public async Task<Connection> ActivateAsync(CancellationToken token)
+        {
+            await Task.Yield();
+
+            var currentAssembly = typeof(RazorLanguageServerClient).Assembly;
+            var currentAssemblyLocation = currentAssembly.Location;
+            var extensionDirectory = Path.GetDirectoryName(currentAssemblyLocation);
+            var languageServerPath = Path.Combine(extensionDirectory, "LanguageServer", "rzls.exe");
+            var info = new ProcessStartInfo();
+            info.FileName = languageServerPath;
+            info.Arguments = "-lsp --logLevel Trace";
+            info.RedirectStandardInput = true;
+            info.RedirectStandardOutput = true;
+            info.UseShellExecute = false;
+            info.CreateNoWindow = true;
+
+            Process process = new Process();
+            process.StartInfo = info;
+
+            if (process.Start())
+            {
+                return new Connection(process.StandardOutput.BaseStream, process.StandardInput.BaseStream);
+            }
+
+            return null;
+        }
+
+        public async Task OnLoadedAsync()
+        {
+            await StartAsync.InvokeAsync(this, EventArgs.Empty);
+        }
+
+        public Task OnServerInitializeFailedAsync(Exception e)
+        {
+            return Task.CompletedTask;
```

Comment:
All three of these functions strike me as in need of diagnostic logging (but especially `OnServerInitializeFailedAsync`). Would it be possible to add a constructor that takes a logger so it can be used here, or will the DI system not enable that?

Reply:
You're definitely right. We'll expand on the implementation of this class in the coming months. For now it's just basic boiler plate to get the ball rolling 😄